### PR TITLE
*: go 1.18 support 🎉😍🎊🍰🥂

### PR DESCRIPTION
**Pull request by AnnaShaleva**
Upgrade minimum required go version and adjust CI workflows.

### CODE REVIEW

File: cli/contract_test.go [hunk 1]
```diff
@@ -578,10 +577,10 @@ func TestComlileAndInvokeFunction(t *testing.T) {
 			e.RunWithError(t, cmd...)
 		})
 		t.Run("corrupted wallet", func(t *testing.T) {
-			tmp, err := ioutil.TempDir("", "tmp")
+			tmp := t.TempDir()
 			require.NoError(t, err)
```
Comment (roman-khimov) [inline cli/contract_test.go:581]: Makes no sense with the change above.

File: pkg/compiler/compiler.go [hunk 2]
```diff
@@ -151,15 +150,15 @@ func getBuildInfo(name string, src interface{}) (*buildInfo, error) {
...
 			if err != nil {
 				return nil, err
 			}
 		default:
 			panic(fmt.Sprintf("unsupported src type: %T", s))
 		}
 		if strings.HasPrefix(runtime.Version(), "go1.15") {
-			dir, err = ioutil.TempDir("", "*")
+			dir = os.TempDir()
 			if err != nil {
```
Comment (roman-khimov) [inline pkg/compiler/compiler.go]: Ditto.

File: pkg/network/tcp_transport.go [hunk 3]
```diff
@@ -78,9 +76,7 @@ func (t *TCPTransport) Accept() {
 
 func (t *TCPTransport) isCloseError(err error) bool {
 	if opErr, ok := err.(*net.OpError); ok {
-		if reClosedNetwork.Match([]byte(opErr.Error())) {
-			return true
-		}
+		return errors.Is(opErr, net.ErrClosed)
```
Comment (fyrchik) [inline pkg/network/tcp_transport.go:79]: What about removing `isCloseError` and just using `errors.Is` at the call-site? Seems quite readable to me.

File: pkg/compiler/compiler.go [hunk 4]
```diff
@@ -157,14 +157,7 @@ func getBuildInfo(name string, src interface{}) (*buildInfo, error) {
 		default:
 			panic(fmt.Sprintf("unsupported src type: %T", s))
 		}
-		if strings.HasPrefix(runtime.Version(), "go1.15") {
```
Comment (fyrchik) [inline pkg/compiler/compiler.go:161]: `runtime` import is now unused, should be removed.

**Comment by roman-khimov** [comment]
At last, https://go.dev/blog/go1.18

File: go.mod [hunk 5]
```diff
@@ -31,4 +31,4 @@ require (
 	gopkg.in/yaml.v2 v2.4.0
 )
 
-go 1.15
+go 1.16
```
Comment (roman-khimov) [inline go.mod:34]: How about `examples/nft-d/go.mod`?

Comment (roman-khimov) [reply go.mod:34]: README.md needs to also be updated,

Comment (AnnaShaleva) [reply go.mod:34]: Finally, I was able to push changes to remote, so fixed.

Comment (AnnaShaleva) [reply go.mod:34]: I'm not sure, is it enought to update gomod of `nft-d` contract or we need to update the `neo-go`'s gomod also?

Comment (AnnaShaleva) [reply go.mod:34]: But how can we squash them? The perpose of the update is to use `neo-go` dependency of `1.16` version everywhere, so it should be a separate commit.

Comment (roman-khimov) [reply go.mod:34]: OK.

Comment (AnnaShaleva) [reply go.mod:34]: Fixed, ready for review.

**Review by roman-khimov — APPROVED** [review]

**Review by fyrchik — APPROVED** [review]

**Pull request merged**